src/ProtoNN/ProtoNNPredictor.cpp

```diff
@@ -38,9 +140,126 @@ ProtoNNPredictor::ProtoNNPredictor(
 #endif
 }
 
+void ProtoNNPredictor::createOutputDirs()
+{
+  std::string subdirName = model.hyperParams.subdirName();
+  outDir = outDir + "/ProtoNNPredictor_" + subdirName;
+
+  try {
+    std::string testcommand = "test -d " + outDir;
+    std::string command = "mkdir " + outDir;
+
+#ifdef LINUX
+    if (system(testcommand.c_str()) == 0)
+      LOG_INFO("Directory " + outDir + " already exists.");
+    else
+      if (system(command.c_str()) != 0)
+        LOG_WARNING("Error in creating directory at this location: " + outDir);
+#endif
+
+#ifdef DUMP
+    testcommand = "test -d " + outDir + "/dump";
+    command = "mkdir " + outDir + "/dump";
+#ifdef LINUX
+    if (system(testcommand.c_str()) == 0)
+      LOG_INFO("Directory " + outDir + "/dump already exists.");
+    else
+      if (system(command.c_str()) != 0)
+        LOG_WARNING("Error in creating directory at this location: " + outDir + "/dump");
+#endif
+#endif
+#ifdef VERIFY
+    testcommand = "test -d " + outDir + "/verify";
+    command = "mkdir " + outDir + "/verify";
+#ifdef LINUX
+    if (system(testcommand.c_str()) == 0)
+      LOG_INFO("Directory " + outDir + "/verify already exists.");
+    else
+      if (system(command.c_str()) != 0)
+        LOG_WARNING("Error in creating directory at this location: " + outDir + "/verify");
+#endif
+#endif
+  }
+  catch (...) {
+    LOG_WARNING("Error in creating one of the subdirectories. Some of the output may not be recorded.");
+  }
+}
+
+
+void ProtoNNPredictor::setFromArgs(const int argc, const char** argv)
```

Comment:
@t-chgupt  If the user does not provide sufficient arguments, do we assert and fail